bin/parse_pom_xml

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env ruby
+
+# Write a JSON string of the dependencies of a Java Maven project
+# to stdout using its pom.xml project file
+
+require 'json'
+
+dependencies_regex = %r{
+    <dependencies>
+        [\s\S]*
+    <\/dependencies>
+}x
+
+dependency_metadata_regex = %r{
+    <dependency>\s*
+        <groupId>
+            (?<groupId>\S+)
+        <\/groupId>\s*
+        <artifactId>
+            (?<artifactId>\S+)
+        <\/artifactId>\s*
+        (<version>
+            (?<version>\S+)
+        <\/version>\s*)?
+        (<scope>
+            (?<scope>\S+)
+        <\/scope>\s*)?
+        (<type>
+            (?<type>\S+)
+        <\/type>\s*)?
+    <\/dependency>
+}x
+
+dependency_lists = File.read(ARGV[0]).scan(dependencies_regex)
+
+dependencies = []
+
+# Get contents within <dependencies> tags
+dependency_lists.each do |dependency_list|
+  # Parse individual <dependency> entries
+  dependency_list.scan(dependency_metadata_regex).each do |dependency_properties|
+    dependency_hash = {}
+    dependency_hash[:group_id] = dependency_properties[0]
+    dependency_hash[:artifact_id] = dependency_properties[1]
+    dependency_hash[:version] = dependency_properties[2]
+    dependency_hash[:scope] = dependency_properties[3]
+    dependency_hash[:type] = dependency_properties[4]
+    dependencies.append(dependency_hash)
+  end
```

Comment:
I did not review the `dependency_lists.each` block because you will likely modify this part to work with an existing xml parser.